src/main/java/com/spotify/docker/client/messages/swarm/Task.java

```diff
@@ -0,0 +1,277 @@
+
+    @JsonProperty("ServiceID")
+    private String serviceId;
+
+    @JsonProperty("Slot")
+    private Integer slot;
+
+    @JsonProperty("NodeID")
+    private String nodeId;
+
+    @JsonProperty("Status")
+    private TaskStatus status;
+
+    @JsonProperty("DesiredState")
+    private String desiredState;
+
+    @JsonProperty("NetworksAttachments")
+    private ImmutableList<NetworkAttachment> networkAttachments;
+
+    public String id() {
+        return id;
+    }
+
+    public Version version() {
+        return version;
+    }
+
+    public Date createdAt() {
+    return createdAt == null ? null : new Date(createdAt.getTime());
+}
+
+    public Date updatedAt() {
+    return updatedAt == null ? null : new Date(updatedAt.getTime());
+}
+
+    public String name() {
+        return name;
+    }
+
+    public Map<String, String> labels() {
+        return labels;
+    }
+
+    public TaskSpec spec() {
+        return spec;
+    }
+
+    public String serviceId() {
+        return serviceId;
+    }
+
+    public Integer slot() {
+        return slot;
+    }
+
+    public String nodeId() {
+        return nodeId;
+    }
+
+    public TaskStatus status() {
+        return status;
+    }
+
+    public String desiredState() {
+        return desiredState;
+    }
+
+    public List<NetworkAttachment> networkAttachments() {
+        return networkAttachments;
+    }
+
+    public static class Criteria {
+
+        /** Filter by task id */
+        String taskId;
+
+        /** Filter by task name */
+        String taskName;
+
+        /** Filter by service name */
+        String serviceName;
+
+        /** Filter by node id */
+        String nodeId;
+
+        /** Filter by label */
+        String label;
+
+        /** Filter by desired state */
+        String desiredState;
+
+        public String getTaskId() {
+            return taskId;
+        }
+
+        public void setTaskId(String taskId) {
+            this.taskId = taskId;
+        }
+
+        public String getTaskName() {
+            return taskName;
+        }
+
+        public void setTaskName(String taskName) {
+            this.taskName = taskName;
+        }
+
+        public String getServiceName() {
+            return serviceName;
+        }
+
+        public void setServiceName(String serviceName) {
+            this.serviceName = serviceName;
+        }
+
+        public String getNodeId() {
+            return nodeId;
+        }
+
+        public void setNodeId(String nodeId) {
+            this.nodeId = nodeId;
+        }
+
+        public String getLabel() {
+            return label;
+        }
+
+        public void setLabel(String label) {
+            this.label = label;
+        }
+
+        public String getDesiredState() {
+            return desiredState;
+        }
+
+        public void setDesiredState(String desiredState) {
+            this.desiredState = desiredState;
+        }
+    }
+
+    public static class CriteriaBuilder {
```

Comment:
For consistency, can you use the builder pattern in `ContainerConfig`? Also make sure the access modifiers for classes and methods are as strict as they can be.